syncapi/streams/stream_pdu.go

```diff
@@ -252,7 +252,13 @@ func (p *PDUStreamProvider) addRoomDeltaToResponse(
 		return r.From, fmt.Errorf("p.DB.RecentEvents: %w", err)
 	}
 	recentEvents := snapshot.StreamEventsToEvents(device, recentStreamEvents)
-	delta.StateEvents = removeDuplicates(delta.StateEvents, recentEvents) // roll back
+	// Don't remove the state events if this is a newly joined room. The member
+	// needs the full room state. For example, to figure out if the room is a space
+	// or not, it needs the m.room.create event which may be chronologically outside
+	// the filter limit.
+	if !delta.NewlyJoined {
```

Comment:
This isn't the right way to approach this I don't think. The idea here is that an event should appear in *either* `"timeline"` or `"state"` but not both.

I suspect this is just a case of us calling `removeDuplicates` at the wrong time, i.e. before filtering rather than afterwards.